tests/services/test_cases.py

```diff
@@ -0,0 +1,119 @@
+from py42.services.cases import CasesService
+
+DUMMY_GET_ALL_TEST_RESPONSE = {
+    "cases": [
+        {
+            "assignee": "string",
+            "assigneeUsername": "string",
+            "createdAt": "2021-01-04T08:09:58.832Z",
+            "createdByUserUid": "string",
+            "createdByUsername": "string",
+            "lastModifiedByUserUid": "string",
+            "lastModifiedByUsername": "string",
+            "name": "string",
+            "number": 0,
+            "status": "OPEN",
+            "subject": "string",
+            "subjectUsername": "string",
+            "updatedAt": "2021-01-04T08:09:58.832Z",
+        }
+    ],
+    "totalCount": 0,
+}
+
+
+class TestCasesService:
+    def test_create_called_with_expected_url_and_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        cases_service.create(
+            u"name", u"subject", u"user uid", u"description", u"findings"
+        )
+        assert mock_connection.post.call_args[0][0] == u"/api/v1/case"
+        print(mock_connection.post.call_args[0][1])
+        post_data = mock_connection.post.call_args[0][1]
+        assert (
+            post_data["name"] == u"name"
+            and post_data["subject"] == u"subject"
+            and post_data["assignee"] == u"user uid"
+            and post_data["description"] == u"description"
+            and post_data["findings"] == u"findings"
+        )
+
+    def test_get_all_called_with_expected_url_and_default_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.side_effect = [
+            DUMMY_GET_ALL_TEST_RESPONSE,
+            DUMMY_GET_ALL_TEST_RESPONSE,
+        ]
+        for _ in cases_service.get_all():
+            continue
+
+        mock_connection.get.call_count == 2
+        assert (
+            mock_connection.get.call_args[0][0]
+            == u"/api/v1/case?pgNum=1&pgSize=500&srtDir=asc&srtKey=number"
+        )
+
+    def test_get_all_called_with_expected_url_and_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.side_effect = [
```

Comment:
I generally prefer mock generators are actually generators because it is had bitten me in the past by them not being so.

There two data types _are_ different, even though they are both iterable.

Reply:
Oops, I thought this was a get-all method, disregard

Reply:
`        items = [
            mock_case_response,
            mock_case_response,
        ]
        def gen():
            yield(next(items))